Services/Masterdata/Services.Masterdata.EF/AutomapperProfile.cs

```diff
@@ -11,11 +14,92 @@ public AutomapperProfile()
         // Masterdata
         CreateMap<MasterdataType, MasterdataTypeListModel>();
         CreateMap<MasterdataType, MasterdataTypeModel>();
-        CreateMap<MasterdataTypeCreateModel, MasterdataType>();
-        CreateMap<MasterdataTypeUpdateModel, MasterdataType>();
+        CreateMap<MasterdataTypeCreateModel, MasterdataType>()
+            .AfterMap(UpdateMetadata);
+        CreateMap<MasterdataTypeUpdateModel, MasterdataType>()
+            .AfterMap(UpdateMetadata);
 
         CreateMap<Entities.Masterdata, MasterdataModel>();
         CreateMap<MasterdataCreateModel, Entities.Masterdata>();
         CreateMap<MasterdataUpdateModel, Entities.Masterdata>();
     }
+
+    private void UpdateMetadata(IMetadataModel source, MasterdataType destination)
+    {
+        // if nothing was passed in, or we have no clue on the domain, we do nothing with the metadata
+        if (source.Metadata is null || string.IsNullOrEmpty(source.Domain)) return;
+
+        Dictionary<string, dynamic>? finalMetadataDictionary = null;
+
+        // if the database already contains metadata
+        if (destination?.MetadataJson != null) 
+        {
+            finalMetadataDictionary = JsonSerializer.Deserialize<Dictionary<string, dynamic>>(destination.MetadataJson);
+            if (finalMetadataDictionary != null)
+            {
+                // We assume you passed in a complete metadata object (dictionary<string, dynamic)
+                // We merge this nicely with the given metadata
+                if (source.Domain == IMetadataModel.AllDomains)
+                {
+                    // for each passed in keyValue
+                    string sourceMetadataJson = source.Metadata.ToString();
```

Comment:
Different variable notations used, sometimes var, sometimes the actual type. I should use var everywhere to make refactoring a bit easier in some cases

Reply:
@jstorteboom notification!